### PR TITLE
New PR Final CR FIxes

### PR DESCRIPTION
**Pull request by RudeWolfCoding**
Test from dashboard-fixed on the implementation

### CODE REVIEW

**Comment by RudeWolfCoding** [comment]
The PR seem to bring no conflict and is ready for merge. Shall I discard the other PR  and just merge this PR?

File: src/components/Rewards/index.tsx [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+  line-height: 3rem;
+  border-bottom: 1px solid black;
+`
+const SupplyField = styled('div')`
+  flex: 1 1 10%;
+  line-height: 3rem;
+  border-bottom: 1px solid black;
+`
+
+// eslint-disable-next-line react/display-name
```
Comment (mul53) [inline src/components/Rewards/index.tsx:98]: remove this comment

File: src/components/Rewards/index.tsx [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+export default function RewardsComponent() {
```
Comment (mul53) [inline src/components/Rewards/index.tsx:99]: rename to Rewards

File: src/components/Rewards/reward.tsx [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+  border-radius: 12px;
+  text-align: center;
+  margin: auto;
+  position: absolute;
+  top: 25%;
+  right: 15px;
+  color: black;
+`
+
+export default function RewardItem(props: { data: any; active: boolean }) {
```
Comment (mul53) [inline src/components/Rewards/reward.tsx:125]: change to Reward

File: src/utils/rewards/index.ts [hunk 4]
```diff
@@ -0,0 +1,76 @@
+import Staking from '../../constants/abis/stakeMethods.json'
+import ethers from 'ethers'
+import { formatEther, parseUnits } from 'ethers/lib/utils'
+import { getProviderOrSigner } from '..'
+import BasicTokenABI from '../../constants/abis/tokenABI.json'
+
+const url = 'https://rpc.fuse.io'
```
Comment (mul53) [inline src/utils/rewards/index.ts:7]: we already store this in a variable called NETWORK_URL

File: src/hooks/useDeprecated.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  const tokens = useAllSwapTokens()
+  const balance = useTokenBalanceCallback()
+  const [isDeprecated, setDeprecated] = useState(false)
+  const [tokenDep, setTokenDep] = useState<Partial<WrappedTokenInfo>>()
+  const [currencyDep, setCurrencyDep] = useState<Currency>()
+  const [balanceDep, setBalance] = useState(0)
+  useEffect(() => {
+    if (account) {
+      console.log('hookstart')
+      Object.keys(tokens).forEach(async key => {
```
Comment (mul53) [inline src/hooks/useDeprecated.ts:19]: does this work well with multiple deprecated tokens?

File: src/pages/App.tsx [hunk 6]
```diff
@@ -1,98 +1,103 @@
...
-      padding: 16px;
-  `};
-
-  z-index: 1;
+  position: relative;
 `
 
-const Marginer = styled.div`
-  margin-top: 5rem;
+const Zindex = styled.div`
```
Comment (mul53) [inline src/pages/App.tsx:58]: why is this called zIndex

File: src/constants/rewards.ts [hunk 7]
```diff
@@ -0,0 +1,54 @@
+export interface Reward {
```
Comment (mul53) [inline src/constants/rewards.ts:1]: hmmmm this shouldn't be in constants

File: src/components/Rewards/withdraw.tsx [hunk 8]
```diff
@@ -0,0 +1,225 @@
...
+    setEstimate(rewards.toFixed(2).toString())
+    setWithdrawValue(calculated)
+  }
+
+  useEffect(() => {
+    setEstimate(props.data.rewardEstimate)
+    setWithdrawValue(0)
+    setContract(props.contract)
+    setResult(props.data)
+  }, [props])
```
Comment (mul53) [inline src/components/Rewards/withdraw.tsx:153]: better to de structure this so that the component re-renders only when certain properties change

**Review by mul53 — CHANGES_REQUESTED** [review]

File: src/components/Header/index.tsx [hunk 9]
```diff
@@ -193,57 +120,51 @@ export const NETWORK_LABELS: any = {
   [BINANCE_MAINNET_CHAINID]: 'Binance'
 }
 
+function accounts(account: any, userEthBalance: any, chainId: any) {
```
[the remark this answers is not in the document]
Comment (leonprou) [reply src/components/Header/index.tsx:123]: let's skip that one for now

File: src/components/HomeCards/prices.tsx [hunk 10]
```diff
@@ -0,0 +1,182 @@
...
+  width: 100%;
+  color: #b5b9d3;
+  font-style: normal;
+`
+
+export default function HomePrices() {
+  const [responseData, setResponseData] = React.useState('0.00')
+  const [statsData, setStats] = React.useState({ pairCount: '0', totalLiquidityUSD: '0', totalVolumeUSD: '0' })
+
+  const props = useSpring({ val: Number(parseFloat(responseData).toFixed(2)), from: { val: 0 } })
```
[the remark this answers is not in the document]
Comment (leonprou) [reply src/components/HomeCards/prices.tsx:102]: let's skip that one for now

File: src/components/Rewards/withdraw.tsx (hunk 8 above)
Comment (leonprou) [reply src/components/Rewards/withdraw.tsx:153]: let's skip that one for now

File: src/constants/rewards.ts (hunk 7 above)
Comment (leonprou) [reply src/constants/rewards.ts:1]: yep

File: src/pages/App.tsx (hunk 6 above)
Comment (leonprou) [reply src/pages/App.tsx:58]: yep. should be called `fullscreen` or something

File: src/utils/farm/index.ts [hunk 11]
```diff
@@ -159,7 +159,7 @@ export async function calculateAPY(
     token1Pool: number
   }
 ) {
-  const url = 'https://rpc.fuse.io'
```
Comment (mul53) [inline src/utils/farm/index.ts]: https://github.com/fuseio/fuseswap-interface/blob/74e6e8c24d171dfab0fc34c956f6aea656f10ab8/src/connectors/index.ts#L11

**Review by mul53 — CHANGES_REQUESTED** [review]

**Review by mul53 — APPROVED** [review]

**Pull request merged**